domain/src/test/java/org/mobilitydata/gtfsvalidator/domain/entity/timeconversionutils/TimeConversionUtilsTest.java

```diff
@@ -0,0 +1,185 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.mobilitydata.gtfsvalidator.domain.entity.timeconversionutils;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+class TimeConversionUtilsTest {
+    private static final int HOUR_TO_SEC_CONVERSION_FACTOR = 3600;
+    private static final int NOON = (12 * HOUR_TO_SEC_CONVERSION_FACTOR);
+    private static final int MIN_TO_SEC_CONVERSION_FACTOR = 60;
+    private static final int SEC_TO_SEC_CONVERSION_FACTOR = 1;
+
+    // Remove warning "PointlessArithmeticExpression" since they these expressions are written to ease code
+    // comprehension
+    @SuppressWarnings("PointlessArithmeticExpression")
+    @Test
+    void hourInHMMSSShouldConvertToIntegerFromNoon() {
+        final String threePm = "15:00:00";
+        int toCheck = TimeConversionUtils.convertHHMMSSToIntFromNoonOfDayOfService(threePm);
+        assertEquals((15*HOUR_TO_SEC_CONVERSION_FACTOR +
+                0*MIN_TO_SEC_CONVERSION_FACTOR + 0*SEC_TO_SEC_CONVERSION_FACTOR) - NOON, toCheck);
+
+        final String noon = "12:00:00";
+        toCheck = TimeConversionUtils.convertHHMMSSToIntFromNoonOfDayOfService(noon);
+        assertEquals(((NOON + 0*MIN_TO_SEC_CONVERSION_FACTOR + 0*SEC_TO_SEC_CONVERSION_FACTOR) - NOON), toCheck);
+
+        final String midnight = "24:00:00";
+        toCheck = TimeConversionUtils.convertHHMMSSToIntFromNoonOfDayOfService(midnight);
+        assertEquals(((24*HOUR_TO_SEC_CONVERSION_FACTOR + 0*MIN_TO_SEC_CONVERSION_FACTOR +
+                0*SEC_TO_SEC_CONVERSION_FACTOR) - NOON), toCheck);
+
+        final String sixFortyPm = "18:40:00";
+        toCheck = TimeConversionUtils.convertHHMMSSToIntFromNoonOfDayOfService(sixFortyPm);
+        assertEquals(((18*3600 + 40*60 + 0) - (12*3600)), toCheck);
+
+        final String tenAm = "10:00:00";
+        toCheck = TimeConversionUtils.convertHHMMSSToIntFromNoonOfDayOfService(tenAm);
+        assertEquals(((10*HOUR_TO_SEC_CONVERSION_FACTOR + 0*MIN_TO_SEC_CONVERSION_FACTOR +
+                0*SEC_TO_SEC_CONVERSION_FACTOR) - NOON), toCheck);
+
+        final String sixThirtyAm20sec = "06:30:20";
+        toCheck = TimeConversionUtils.convertHHMMSSToIntFromNoonOfDayOfService(sixThirtyAm20sec);
+        assertEquals(((6*HOUR_TO_SEC_CONVERSION_FACTOR + 30*MIN_TO_SEC_CONVERSION_FACTOR +
+                20*SEC_TO_SEC_CONVERSION_FACTOR) - NOON), toCheck);
+
+        final String oneThirtyAm40sec = "25:30:40";
+        toCheck = TimeConversionUtils.convertHHMMSSToIntFromNoonOfDayOfService(oneThirtyAm40sec);
+        assertEquals(((25*HOUR_TO_SEC_CONVERSION_FACTOR + 30*MIN_TO_SEC_CONVERSION_FACTOR +
+                40*SEC_TO_SEC_CONVERSION_FACTOR) - NOON), toCheck);
+
+         // check that null value returns null when calling method
```

Comment:
this shall be split into its own testing method with explicit name. That way, reading test names informs about the API behavior